core/src/test/java/org/bitcoinj/wallet/WalletTest.java

```diff
@@ -3441,4 +3442,39 @@ public void totalReceivedSent() throws Exception {
 
         // TODO: test shared wallet calculation here
     }
+
+    @Test
+    public void testIrrelevantDoubleSpend() throws Exception {
+        // 1 spends 0
+        Transaction tx0 = createFakeTx(PARAMS, COIN.multiply(5), OTHER_ADDRESS, OTHER_ADDRESS2)[1];
```

Comment:
This version of createFakeTx also creates feeder transactions which you throw away. Is this needed? Probably we could use the "one address" version here?